pkg/azclient/cloud.go

```diff
@@ -123,61 +136,35 @@ func AzureCloudConfigOverrideFromEnv(config *cloud.Configuration) (*cloud.Config
 			Audience: envConfig.TokenAudience,
 		}
 	}
-	return config, nil
+	if len(envConfig.StorageEndpointSuffix) > 0 {
+		armconfig.StorageSuffix = &envConfig.StorageEndpointSuffix
+	}
+	if len(envConfig.ContainerRegistryDNSSuffix) > 0 {
+		armconfig.ACRLoginServer = &envConfig.ContainerRegistryDNSSuffix
+	}
+	return nil
 }
 
 // GetAzureCloudConfig returns the cloud configuration for the given ARMClientConfig.
-func GetAzureCloudConfig(armConfig *ARMClientConfig) (*cloud.Configuration, error) {
+func BackfillAzureCloudConfig(armConfig *ARMClientConfig) (*cloud.Configuration, error) {
```

Comment:
why it's renamed here? can you provide more comments for this func?

Reply:
The function will change the content of parameter. GetXxx is not a proper name because it indicates that the parameter will not change

Reply:
renamed to GetAzureCloudConfigAndBackfillARMClientConfig and added some comments.